### PR TITLE
Add manager and agent script

### PR DESCRIPTION
**Pull request by bk201**
- The go program `support-bundle-utils` has a `manager` command to start the support bundle manager.
  The manager does these works: 
    - It collects cluster bundle, including YAML manifests and pod logs.
    - It collects external bundles. e.g., Longhorn support bundle.
    - It starts a web server and wait for bundle downloading and uploading.
    - It starts a daemonset on each node. The agents in the daemonset collect node bundles and push them back to the manager.
- The script `harvester-sb-collector.sh` collect node bundle.

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

### CODE REVIEW

**Comment by bk201** [comment]
test image: `bk201z/support-bundle-utils:043df39-amd64`

**Comment by bk201** [comment]
test image: `bk201z/support-bundle-utils:3b71108-amd64`

**Comment by gitlawr** [comment]
Can you split go module and vendor changes to a separate commit? It makes code reviewing easier.

**Comment by bk201** [comment]
> Can you split go module and vendor changes to a separate commit? It makes code reviewing easier.

Done. Updated testing image: `bk201z/support-bundle-utils:ecdb3bd-amd64`

**Comment by bk201** [comment]
Add more enhancements.
`bk201z/support-bundle-utils:354bea9-amd64`

File: bin/harvester-sb-collector.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# get some host information
+cp ${HOST_PATH}/etc/hostname .
+
+# collect logs
+mkdir -p logs
+cd logs
+dmesg &> dmesg.log
+
+# k3s logs don't rorate well and can be huge
+tail -c 100m ${HOST_PATH}/var/log/k3s-service.log > k3s-service.log
```
Comment (gitlawr) [inline bin/harvester-sb-collector.sh:27]: Can we change it to 10m? 100m*N(nodes) can exceed github limit so community users cannot upload it on github in most cases.

File: pkg/manager/manager.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	if node == "" {
+		return
+	}
+
+	logrus.Debugf("handle create node bundle for %s", node)
+	nodesDir := filepath.Join(m.getWorkingDir(), "nodes")
+	err := os.MkdirAll(nodesDir, os.FileMode(0775))
+	if err != nil {
+		logrus.Errorf("fail to create directory %s: %s", nodesDir, err)
+		return
```
Comment (gitlawr) [inline pkg/manager/manager.go:233]: Is it expected that when an error happens we returns a statusOK

Comment (bk201) [reply pkg/manager/manager.go]: [Fixed](d419fd6). Thanks!

File: bin/harvester-sb-collector.sh (hunk 1 above)
Comment (bk201) [reply bin/harvester-sb-collector.sh]: [Reduced to 10m.](b7d3b4f)

**Comment by bk201** [comment]
`bk201z/support-bundle-utils:aae0565-amd64`

**Comment by gitlawr** [comment]
LGTM

**Comment by bk201** [comment]
Squashed and update testing image: `bk201z/support-bundle-utils:7072271-amd64`

**Review by gitlawr — APPROVED** [review]

**Review by guangbochen — APPROVED** [review]

**Pull request merged**